src/argilla/client/sdk/v1/datasets/api.py

```diff
@@ -234,6 +234,26 @@ def add_records(
     """
     url = "/api/v1/datasets/{id}/records".format(id=id)
 
+    active_user_id = None
+
+    for record in records:
+        cleaned_responses = []
+        response_without_user_id = False
+        for response in record["responses"]:
+            if response["user_id"] is None:
+                if response_without_user_id is True:
+                    warnings.warn(
```

Comment:
@tomaarsen I believe we opted for using logger.warning() right? I don't fully recall the discussion.

Reply:
Oh cool, then @tomaarsen if you confirm this I'll change this before merging, as well as the `pytest.warns` statements in the unit tests

Reply:
I think we discussed it a bit in #2456. However, I think in situations where you're not providing a special type of warning (e.g. DeprecationWarning), then I think logger.warning might suit Argilla best? In the end, we would just have to normalize throughout the full project.

Reply:
Ok for the moment I'm happy to make it `logger.warning` and we can come back to that issue once we release Argilla 1.8.0?

Reply:
Sounds like the best approach